spec.html

```diff
@@ -709,32 +689,171 @@
+        <p>Algorithms steps that say or are otherwise equivalent to:</p>
+        <emu-alg>
+          1. Let _result_ be AbstractOperation(ReturnIfAbrupt(_argument_)).
+        </emu-alg>
+        <p>mean the same thing as:</p>
+        <emu-alg>
+          1. If _argument_ is an abrupt completion, return _argument_.
+          1. If _argument_ is a Completion Record, let _argument_ be _argument_.[[Value]].
+          1. Let _result_ be AbstractOperation(_argument_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-returnifabrupt-shorthands">
+        <h1>ReturnIfAbrupt Shorthands</h1>
+        <p>Abstract operations referenced using the functional application style and the method application style that are prefixed by `?` indicate that ReturnIfAbrupt should be applied to the resulting Completion Record. For example, the algorithm:</p>
+        <emu-alg>
+          1. ? operationName().
+        </emu-alg>
+        <p>is equivalent to the following algorithm:</p>
+        <emu-alg>
+          1. ReturnIfAbrupt(operationName()).
+        </emu-alg>
+        <p>Similarly, for method application style:</p>
+        <emu-alg>
+          1. ? _someValue_.operationName().
+        </emu-alg>
+        <p>is equivalent to the following algorithm:</p>
+        <emu-alg>
+          1. ReturnIfAbrupt(_someValue_.operationName()).
+        </emu-alg>
+        <p>Similarly, prefix `!` is used to indicate that an abstract operation will never return an abrupt completion and that the resulting Completion Record's value field should be used in place of the return value of the operation. For example, the algorithm:</p>
+        <emu-alg>
+          1. Let _val_ be ! operationName().
+        </emu-alg>
+        <p>is equivalent to the following algorithm steps:</p>
+        <emu-alg>
+          1. Let _val_ be operationName().
+          1. Assert: _val_ is never an abrupt completion.
+          1. If _val_ is a Completion Record, let _val_ be _val_.[[Value]].
+        </emu-alg>
+        <p>Production-associated algorithms for runtime semantics may also make use of this short-hand by placing `!` or `?` before the algorithm's name:</p>
+        <emu-alg>
+          1. Perform ! ProductionAssociatedAlgorithm of |NonTerminal|.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+    <!-- es6num="5.3" -->
+    <emu-clause id="sec-static-semantic-rules">
+      <h1>Static Semantics</h1>
+      <p>Context-free grammars are not sufficiently powerful to express all the rules that define whether a stream of input elements form a valid ECMAScript |Script| or |Module| that may be evaluated. In some situations additional rules are needed that may be expressed using either ECMAScript algorithm conventions or prose requirements. Such rules are always associated with a production of a grammar and are called the <dfn>static semantics</dfn> of the production.</p>
```

Comment:
Having all references be back-references is definitely ideal (part of this change pulls some later clauses in earlier for that reason), although it's not always possible e.g. in cases of circularity. If you come across misordered things though it's an easy fix and PR to make!

I think this PR is pretty reasonable as far as forward references go.